kuzzleio/kuzzle-plugin-cluster
# Kzl 998 strategies in redis

## What does this PR do ?

This PR is the fixed version of https://github.com/kuzzleio/kuzzle-plugin-cluster/pull/74 which, in its original version, would remove the strategies loaded at Kuzzle start -_-;....

### How should this be manually tested?

1. check the local strategy is still working, for instance by using the admin console
2. test "on-the-flight" strategies addon and removal:
  - Add this small plugin to your cluster: https://gist.github.com/benoitvidis/a5d9bd592a261d7b903209e73abed59d
  - Check the `dummy` strategy does not exist by trying to log in
  - Call http://localhost:7512/_plugin/test/do/bar
  - Check the `dummy` strategy is known on all nodes
  - Call http://localhost:7512/_plugin/test/do/foo
  - Check the `dummy` strategy has been deleted from all nodes

Additionally, you can run kuzzle functional tests:
1. switch Kuzzle version to https://github.com/kuzzleio/kuzzle/pull/1284
2. from kuzzle directory, run `./node_modules/.bin/cucumber-js --profile websocketNoRedis`

### Boyscout

dependencies upgrade.
